tools/src/dbgen/Lib.hs

```diff
@@ -30,38 +35,57 @@ import           Pos.Wallet.Web.ClientTypes
 import           Pos.Wallet.Web.ClientTypes.Instances ()
 import           Pos.Wallet.Web.Methods.Logic
 import           Pos.Wallet.Web.Methods.Restore
+import           Pos.Wallet.Web.State.State           (getWalletUtxo, setWalletUtxo, updateWalletBalancesAndUtxo)
+import           Test.QuickCheck                      (arbitrary, generate)
 import           Rendering
 import           Text.Printf
 
 --
 -- Types
 --
 
-data GenSpec = GenSpec {
-    wallets       :: !Integer
+data GenSpec = GenSpec
+    { wallets     :: !Integer
     -- ^ How many wallets to create
     , wallet_spec :: WalletSpec
     -- ^ The specification for each wallet.
     } deriving (Show, Eq, Generic)
 
 instance Interpret GenSpec
 
-data WalletSpec = WalletSpec {
-    accounts       :: !Integer
+data WalletSpec = WalletSpec
+    { accounts      :: !Integer
     -- ^ How many accounts to generate
-    , account_spec :: AccountSpec
+    , account_spec  :: AccountSpec
     -- ^ How specification for each account.
+    , fakeUtxo      :: FakeUtxoSpec
+    -- ^ Configuration for the generation of the fake UTxO.
     } deriving (Show, Eq, Generic)
 
 instance Interpret WalletSpec
 
-data AccountSpec = AccountSpec {
-    addresses :: !Integer
-    -- How many addresses to generate.
+data AccountSpec = AccountSpec
+    { addresses :: !Integer
+    -- ^ How many addresses to generate.
     } deriving (Show, Eq, Generic)
 
 instance Interpret AccountSpec
 
+-- TODO(ks): The question here is whether we need to support some other
+-- strategies for distributing money, like maybe using a fixed amount
+-- of `toAddress` to cap the distribution - we have examples where we
+-- have around 80 000 addresses which is a lot and can be intensive?
+-- For now, KISS.
+data FakeUtxoSpec = FakeUtxoSpec
+    {  fromAddress  :: Maybe Integer
```

Comment:
I find `fromAddress` not very much self-explanatory, and it's hard to evince from the Dhall config. Furthermore, it feels like there is a bit of duplication in the utxo spec, as the overall spec already specify how many addresses to create, we just need to specify how many fill with fake utxo, and with which balance. I propose something like this:

``` haskell
data FakeUtxoSpec = FakeUtxoSpec {
  amount :: !Integer
  -- ^ If set to zero, this effectively corresponds to "do not generate fake utxo".
  , distribution :: CoinDistributionSpec
}

data CoinDistributionSpec = 
    UniformDistribution
    -- ^ All the addresses gets the same amount.
  | RangeDistrubution !Integer
  -- ^ Distributes to only XX addresses.
  | ...
  -- ^ For now we KISS, later we can add more type constructors

instance Interpreted CoinDistributionSpec where
  -- This will need to be written in order to map an object into a proper CoinDistrubitionSpec
```

This way, you can now specify a uniform utxo distribution this way:

```
{ wallet_spec = { account_spec = { addresses = 100 }, accounts = 1, fakeUtxo = { distribution = { type = "uniform" } , amount = 10000 } }
 , wallets     = 1
}
```

But you can now also do the equivalent of the `fromAddresses = [100]` like before by setting `distribution` to be, for example:

```
{ distribution = { type = "range"
                          , addresses = 100
                          }
}
```

It's now easy to add extra policies and we don't need any `Optional`: merely setting the `amount` to 0 or the `addresses` to 0 would entail "Do not distribute stake". What do you think?

Reply:
I tried to avoid that case because it's not as explicit as the `Optional` one - the user must know to set the number to `0` which doesn't make sense to me - imagine if you had to know how to set up a full configuration of a library you don't know.
So that's why my preference is/was:
1. Explicit argument sent to the application (high visibility, low chance of doing something wrong)
2. Explicit configuration parameter (medium visibility, medium chance of doing something wrong)
3. Set the parameter to specific value (low visibility, high chance of doing something wrong)

I guess the other option which would be ok for would be to have (`NoDistribution`):
```
data CoinDistributionSpec = 
    NoDistribution
    -- ^ Do not distribute the coins.
 |  UniformDistribution
    -- ^ All the addresses gets the same amount.
  | RangeDistrubution !Integer
  -- ^ Distributes to only XX addresses.
  | ...
  -- ^ For now we KISS, later we can add more type constructors
```

@adinapoli-iohk Is that ok? 
You can still do something bad with `addresses = 0` but at least you know that you explicitly asked for the generation of the fake utxo.

Reply:
I see your point -- I don't think this in practice would be too much of a concern as this tool is mostly for internal use, and the README should cover the ambiguity. However, I like the `NoDistribution` idea, so I'm happy to settle on this middle ground 😉 

Generally speaking, I don't see necessarily setting some values to 0 as being an evil thing -- it's just another declination of "what you see is what you get", in my opinion. Setting the `amount` to 0 explicitly declares you don't want those addresses to have balance, which is fine. Same applies for `addresses = 0`: you are explicitly stating you want to distribute the stake to exactly 0 addresses. Again, that's your choice, and I'm not sure what `Optional` buys you in that case, but this is more design bikeshedding than anything 😉